lib/miq_expression/field.rb

```diff
@@ -26,7 +26,10 @@ def date?
   end
 
   def self.valid_field?(field)
-    FIELD_REGEX.match(field).present? if field.kind_of?(String)
+    if field.kind_of?(String)
+      match = FIELD_REGEX.match(field)
+      match.present? && match[:model_name].safe_constantize.present?
+    end
```

Comment:
A string value is still valid so the name of the method is not accurate. I'll rename it to `is_field?`